### ISSUE TITLE
CA1305 is still producing false positives

### ISSUE THREAD

**Issue by halter73** [dotnet/roslyn-analyzers#6746]
### Analyzer
**Diagnostic ID**: [CA1305](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1305): `The behavior of 'long?.GetValueOrDefault()' could vary based on the current user's locale settings. Provide a value for the 'IFormatProvider' argument.`

### Analyzer source
**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](8.0.100-preview.7.23356.2)

**Version**: 8.0.100-preview.7.23356.2

_OR_

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 5.0.3 (Latest)

### Describe the bug
The analyzer seems to be producing false positives for calls to GetHashCode() and GetValueOrDefault(). E.g:

```
src\Http\Headers\src\RangeItemHeaderValue.cs(33,31): error CA1305: (NETCORE_ENGINEERING_TELEMETRY=Build) The behavior of 'long?.GetValueOrDefault()' could vary based on the current user's locale settings. Provide a value for the 'IFormatProvider' argument. (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1305)
```

https://dev.azure.com/dnceng-public/public/_build/results?buildId=330777&view=results

![Image](https://user-images.githubusercontent.com/6995051/251519144-02ff1d75-68b4-4c6c-ace3-c0018c8d6559.png)

### Steps To Reproduce
* Clone the branch from
* Checkout https://github.com/dotnet/aspnetcore/commit/802641b8eea572772212858d97c3a753a58df986 from before the workarounds applied in the [Update SDK aspnetcore#49034](https://github.com/dotnet/aspnetcore/pull/49034) PR.
* run eng\build.cmd
* observe the errors.

### Expected behavior
Does not produce false positives.

### Actual behavior
The rule produces some false positives.

## Additional context

This is a dupe of #6745, but I couldn't reopen it and I didn't want to bother @javiercn late at night.

@mavasani @CollinAlpert

**Comment by mavasani** [comment]
Fixed with https://github.com/dotnet/roslyn-analyzers/pull/6748
